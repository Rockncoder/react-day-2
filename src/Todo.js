import React from 'react';

export default class Todo extends React.Component {
  toggleDone = () => {
    this.props.doneChange(this.props.todo.id);
  };

  onClick = e => {
    e.preventDefault();
    this.props.deleteTodo(this.props.todo.id);
  };

  render() {
    const {todo} = this.props;
    const className = todo.done? 'done-todo': '';
    return (
      <div className={`todo ${className} todo-${todo.id}`}>
        <span className="toggle-todo" onClick={this.toggleDone}>
          {todo.name}
        </span>
        <button className="delete-todo" onClick={this.onClick}>Delete</button>
      </div>
    );
  }
}