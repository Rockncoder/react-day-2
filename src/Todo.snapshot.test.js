import Todo from './Todo';
import renderer from 'react-test-renderer';
import React from 'react';

test('Todo component renders the todo correctly', () => {
  const todo = {id: 1, done: false, name: 'Write more code'};
  const rendered = renderer.create(<Todo todo={todo} />);

  expect(rendered.toJSON()).toMatchSnapshot();
});