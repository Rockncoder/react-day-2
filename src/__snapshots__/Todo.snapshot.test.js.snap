// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly 1`] = `
<div
  className="todo  todo-1"
>
  <span
    className="toggle-todo"
    onClick={[Function]}
  >
    Write more code
  </span>
  <button
    className="delete-todo"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
